File: db/view_event_attachment.php
<?php

require_once('view.php');

/**
 * Class for accessing, manipulating, creating and deleting records in event_attachment.
*/

class View_Event_Attachment extends View
{

  public function __construct($select = array())
  {
    $this->table = 'event_attachment, attachment_type';
    $this->order = '';
    $this->domain = 'event';
    $this->join = 'event_attachment.attachment_type_id = attachment_type.attachment_type_id';
    $this->field['event_attachment_id']['type'] = 'SERIAL';
    $this->field['event_attachment_id']['table'] = 'event_attachment';
    $this->field['event_attachment_id']['primary_key'] = true;
    $this->field['attachment_type_id']['type'] = 'INTEGER';
    $this->field['attachment_type_id']['table'] = 'event_attachment';
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['table'] = 'event_attachment';
    $this->field['mime_type_id']['type'] = 'INTEGER';
    $this->field['mime_type_id']['table'] = 'event_attachment';
    $this->field['filename']['type'] = 'VARCHAR';
    $this->field['filename']['table'] = 'event_attachment';
    $this->field['title']['type'] = 'VARCHAR';
    $this->field['title']['table'] = 'event_attachment';
    $this->field['f_public']['type'] = 'BOOL';
    $this->field['f_public']['table'] = 'event_attachment';
    $this->field['data']['type'] = 'BYTEA';
    $this->field['data']['table'] = 'event_attachment';
    $this->field['filesize']['type'] = 'INTEGER';
    $this->field['filesize']['name'] = 'octet_length(event_attachment.data)';
    $this->field['attachment_type_tag']['type'] = 'VARCHAR';
    $this->field['attachment_type_tag']['name'] = 'attachment_type.tag';
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/views/view_event_attachment.rb
module Momomoto
  class View_event_attachment < Base
    def initialize
      super
      @domain = 'event'
      @fields = {
        :event_attachment_id => Datatype::Integer.new( {} ),
        :attachment_type_id => Datatype::Integer.new( {} ),
        :attachment_type => Datatype::Varchar.new( {} ),
        :attachment_type_tag => Datatype::Varchar.new( {:length=>32} ),
        :event_id => Datatype::Integer.new( {} ),
        :mime_type_id => Datatype::Integer.new( {} ),
        :mime_type_tag => Datatype::Varchar.new( {:length=>128} ),
        :mime_type => Datatype::Varchar.new( {:length=>128} ),
        :filename => Datatype::Varchar.new( {:length=>256} ),
        :title => Datatype::Varchar.new( {:length=>256} ),
        :f_public => Datatype::Bool.new( {} ),
        :last_changed => Datatype::Timestamp.new( {} ),
        :filesize => Datatype::Integer.new( {} ),
        :language_id => Datatype::Integer.new( {} )
      }
    end
  end
end
